app/triggers/providers/mqtt/Hass.js

```diff
@@ -124,44 +128,56 @@ class Hass {
         const watcherUpdateStatusDiscoveryTopic = this.getDiscoveryTopic({ kind: 'binary_sensor', topic: watcherUpdateStatusTopic });
 
         // Publish discovery messages
-        await this.publishDiscoveryMessage({
-            discoveryTopic: totalCountDiscoveryTopic,
-            stateTopic: totalCountTopic,
-            name: 'Total container count',
-        });
-        await this.publishDiscoveryMessage({
-            discoveryTopic: totalUpdateCountDiscoveryTopic,
-            stateTopic: totalUpdateCountTopic,
-            name: 'Total container update count',
-        });
-        await this.publishDiscoveryMessage({
-            discoveryTopic: totalUpdateStatusDiscoveryTopic,
-            stateTopic: totalUpdateStatusTopic,
-            name: 'Total container update status',
-            options: {
-                payload_on: true.toString(),
-                payload_off: false.toString(),
-            },
-        });
-        await this.publishDiscoveryMessage({
-            discoveryTopic: watcherTotalCountDiscoveryTopic,
-            stateTopic: watcherTotalCountTopic,
-            name: `Watcher ${container.watcher} container count`,
-        });
-        await this.publishDiscoveryMessage({
-            discoveryTopic: watcherUpdateCountDiscoveryTopic,
-            stateTopic: watcherUpdateCountTopic,
-            name: `Watcher ${container.watcher} container update count`,
-        });
-        await this.publishDiscoveryMessage({
-            discoveryTopic: watcherUpdateStatusDiscoveryTopic,
-            stateTopic: watcherUpdateStatusTopic,
-            name: `Watcher ${container.watcher} container update status`,
-            options: {
-                payload_on: true.toString(),
-                payload_off: false.toString(),
-            },
-        });
+        if (this.configuration.hass.discovery) {
```

Comment:
Why not wrapping all the ` this.publishDiscoveryMessage` calls into a single `if (this.configuration.hass.discovery) {}`?

Reply:
Is [this](https://github.com/fmartinou/whats-up-docker/pull/402/commits/94ca029b822f4b54d9fc6fe5990ebb9388e49f34) what you had in mind?

Reply:
It would be nice, if this could be merged. Could you review the modification, please?

Reply:
Thanks for the change regarding the `if` parts👍 

I'm just noticing one last thing:

With your PR, all users who are currently using `HASS` will have to update their conf before upgrading because `WUD_TRIGGER_MQTT{trigger_name}_HASS_DISCOVERY` will now be by default `false` regardless the value of `WUD_TRIGGER_MQTT_{trigger_name}_HASS_ENABLED`

Do you see a use case where someone would like to integrate to HASS but without enabling the automatic discovery?

I'm wondering if we should not automatically toggle `WUD_TRIGGER_MQTT{trigger_name}_HASS_DISCOVERY` to `true` when `WUD_TRIGGER_MQTT_{trigger_name}_HASS_ENABLED` is enabled?

It should be doable to implement that by changing the Joi validation API directly on the line:
```
discovery: this.joi.boolean().default(false),
```
(changing the `false` by an evaluation of the sibling property `hass.enabled` 🤔 )

What do you think about?

Reply:
You're right, breaking the configuration for existing users should be avoided. Your proposal makes absolutely sense. However, as you maybe can see, I have problems to find the right implementation. Could you help me and tell me exactly how I can evaluate the sibling as you describe?

Reply:
I've not tested by myself but the code is using the [Joi library](https://joi.dev/) and Joi seems to support this use case.

I'd say by combining `when` (to make the conditional work) and `ref` (to reference another property) functions.
- https://joi.dev/api/?v=17.13.3#anywhencondition-options
- https://joi.dev/api/?v=17.13.3#refkey-options

I'll try to find some time to test that if you don't succeed finding the right implementation.

And once again; thank you for your contribution!

Reply:
Thank you very much for giving me a push into the right direction. As far as I could test it, it seems to work now as expected: When I only set `WUD_TRIGGER_MQTT_FHEM_HASS_ENABLED=true`, discovery messages are published. When I set `WUD_TRIGGER_MQTT_FHEM_HASS_DISCOVERY=false`, discovery messages are not published. What do you think?